### PR TITLE
Fix test runner error reporting, which regressed in #6701

### PR DESCRIPTION
**Pull request by kripken**
We forgot to return the error code, so errors were shown on the console but CI didn't notice them and things looked green.

I guess we need a test that checks that errors are noticed by CI, but how would we test it? ;) A question for the philosophers perhaps.

Also this PR fixes a minor regression that occurred meanwhile, in the metadce test.

### CODE REVIEW

**Comment by kripken** [comment]
CI shows we have a few more regressions on incoming because of that issue.

I say we land this with the errors - this just makes them noticeable. Landing it quickly will prevent more from being added. Any objections?

**Comment by kripken** [comment]
Hopefully those commits fix all recent regressions.

File: tests/runner.py [hunk 1]
```diff
@@ -1001,9 +1001,10 @@ def btest(self, filename, expected=None, reference=None, force_c=False, referenc
     filepath = path_from_root('tests', filename) if not filename_is_src else ('main.c' if force_c else 'main.cpp')
     temp_filepath = os.path.join(self.get_dir(), os.path.basename(filepath))
     original_args = args[:]
-    if os.environ.get('EMCC_TEST_WASM_PTHREADS', '0') != '1':
-      # Browsers currently have wasm threads off by default, so don't test them unless explicitly enabled.
-      args = args + ['-s', 'WASM=0']
+    if 'USE_PTHREADS=1' in args or 'USE_PTHREADS=2' in args:
```
Comment (dschuff) [inline tests/runner.py]: lol oops.

File: tests/test_browser.py [hunk 2]
```diff
@@ -3542,13 +3542,15 @@ def test_separate_asm(self):
     for opts in [['-O0'], ['-O1'], ['-O2'], ['-O2', '--closure', '1']]:
       print(opts)
       open('src.cpp', 'w').write(self.with_report_result(open(path_from_root('tests', 'browser_test_hello_world.c')).read()))
-      Popen([PYTHON, EMCC, 'src.cpp', '-o', 'test.html', '-s', 'WASM=0'] + opts).communicate()
+      run_process([PYTHON, EMCC, 'src.cpp', '-o', 'test.html', '-s', 'WASM=0'] + opts)
       self.run_browser('test.html', None, '/report_result?0')
 
+      print('run one');
```
Comment (dschuff) [inline tests/test_browser.py]: were these debug printfs?

Comment (kripken) [reply tests/test_browser.py]: Yeah, but it's nice to keep them, I think - they show progress through the test, if you're watching it run.

Comment (jgravelle-google) [reply tests/test_browser.py]: Having debugged several tests that do this in test_core, they also help pinpoint which iteration of the test failed.

File: tools/separate_asm.py [hunk 3]
```diff
@@ -25,6 +25,10 @@
   # seek a pattern like (e.ENVIRONMENT), which is in the shell.js if-cascade for the ENVIRONMENT override
   import re
   m = re.search('\((\w+)\.ENVIRONMENT\)', everything)
+  if not m:
+    m = re.search('(\w+)\.arguments = \[\];', everything)
+  if not m:
+    m = re.search('(\w+)\.arguments=\[\];', everything)
```
Comment (jgravelle-google) [inline tools/separate_asm.py]: Am I reading this right and it's saying match `xxx.arguments = [];`, with or without whitespace around the `=`?
How about merging these into something like
`m = re.search('(\w+)\.arguments\s*=\s*\[\];', everything)`

Comment (kripken) [reply tools/separate_asm.py]: Good idea, done.

**Pull request merged**